traversal/scanner/GraphIterator.java

```diff
@@ -194,7 +193,7 @@ private void setCompleted() {
     }
 
     private void initialiseStarts() {
-        toTraverse.addAll(procedure.startVertices());
+        toTraverse.add(procedure.vertex(0));
```

Comment:
since we always proceed to the next ordered vertex, we can just start at the 0th vertex now